bugsnag-android-core/src/main/java/com/bugsnag/android/Configuration.java

```diff
@@ -0,0 +1,833 @@
+     *
+     * In an android app the "context" is automatically set as the foreground Activity.
+     * If you would like to set this value manually, you should alter this property.
+     */
+    public void setContext(@Nullable String context) {
+        impl.setContext(context);
+    }
+
+    /**
+     * Sets which values should be removed from any Metadata objects before
+     * sending them to Bugsnag. Use this if you want to ensure you don't send
+     * sensitive data such as passwords, and credit card numbers to our
+     * servers. Any keys which contain these strings will be filtered.
+     *
+     * By default, redactedKeys is set to "password"
+     */
+    @Nullable
+    public Set<String> getRedactedKeys() {
+        return impl.getRedactedKeys();
+    }
+
+    /**
+     * Sets which values should be removed from any Metadata objects before
+     * sending them to Bugsnag. Use this if you want to ensure you don't send
+     * sensitive data such as passwords, and credit card numbers to our
+     * servers. Any keys which contain these strings will be filtered.
+     *
+     * By default, redactedKeys is set to "password"
+     */
+    public void setRedactedKeys(@Nullable Set<String> redactedKeys) {
+        if (redactedKeys != null) {
+            impl.setRedactedKeys(redactedKeys);
+        } else {
+            impl.setRedactedKeys(Collections.<String>emptySet());
+        }
+    }
+
+    /**
+     * Allows you to specify the fully-qualified name of error classes that will be discarded
+     * before being sent to Bugsnag if they are detected. The notifier performs an exact
+     * match against the canonical class name.
+     */
+    @Nullable
+    public Set<String> getDiscardClasses() {
+        return impl.getDiscardClasses();
+    }
+
+    /**
+     * Allows you to specify the fully-qualified name of error classes that will be discarded
+     * before being sent to Bugsnag if they are detected. The notifier performs an exact
+     * match against the canonical class name.
+     */
+    public void setDiscardClasses(@Nullable Set<String> discardClasses) {
+        if (discardClasses != null) {
+            impl.setDiscardClasses(discardClasses);
+        } else {
+            impl.setDiscardClasses(Collections.<String>emptySet());
+        }
+    }
+
+    /**
+     * By default, Bugsnag will be notified of events that happen in any releaseStage.
+     * If you would like to change which release stages notify Bugsnag you can set this property.
+     */
+    @Nullable
+    public Set<String> getEnabledReleaseStages() {
+        return impl.getEnabledReleaseStages();
+    }
+
+    /**
+     * By default, Bugsnag will be notified of events that happen in any releaseStage.
+     * If you would like to change which release stages notify Bugsnag you can set this property.
+     */
+    public void setEnabledReleaseStages(@Nullable Set<String> enabledReleaseStages) {
+        impl.setEnabledReleaseStages(enabledReleaseStages);
```

Comment:
Set is nullable, but not the Strings within it.... do we need to check them (personally I think so, based only what I readon the Kotlin null safety page, even if it it is a nause)?

Reply:
Resolving as this will be covered by the other comment above.